File: src/main/java/com/amarpreet/logparser/data/LogDataSummaryDetail.java
package com.amarpreet.logparser.data;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Value;

import java.util.List;

@Value
@Builder
@AllArgsConstructor
public class LogDataSummaryDetail {
    private String numberOfUnique;
    private List<String> top3Urls;
    private List<String> top3Ips;
}

File: src/models/LogDataSummaryDetail.js
/**
 * LogDataSummaryDetail model - represents the summary of log data
 */
class LogDataSummaryDetail {
  constructor(numberOfUnique, top3Urls, top3Ips) {
    this.numberOfUnique = numberOfUnique;
    this.top3Urls = top3Urls;
    this.top3Ips = top3Ips;
  }
}

module.exports = LogDataSummaryDetail;
